### PR TITLE
MSW fetchResult abstraction

### PR DESCRIPTION
**Pull request by JacobMGEvans**
- As part of the effort regarding reimplementing abstractions and refactoring the work from MSW migration.

 related #2307

Author has included the following, where applicable:

- [ ] Tests
~- [ ] Changeset~

Reviewer has performed the following, where applicable:

- [ ] Checked for inclusion of relevant tests
~- [ ] Checked for inclusion of a relevant changeset~
~- [ ] Checked for creation of associated docs updates~
- [ ] Manually pulled down the changes and spot-tested

### CODE REVIEW

File: packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts [hunk 1]
```diff
@@ -52,25 +49,22 @@ export const mswSuccessDeployments = [
 								},
 							},
 						],
-					} as DeploymentListRes,
-				})
+					} as DeploymentListRes)
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts:52]: I don't think we actually need a cast here, right?

File: packages/wrangler/src/__tests__/helpers/msw/handlers/script.ts [hunk 2]
```diff
@@ -29,46 +30,28 @@ export default [
 	rest.get(
 		"*/accounts/:accountId/workers/services/:scriptName/environments/:env/content",
 		({ params: { scriptName } }, res, context) => {
-			return res(
-				context.status(200),
-				context.text(getScript(scriptName as string))
-			);
+			return res(context.text(getScript(scriptName as string)));
```
Comment (petebacondarwin) [inline packages/wrangler/src/__tests__/helpers/msw/handlers/script.ts:33]: NIT: you could change `getScript()` to accept a `PathParams<string>` and do the cast to string in there, avoiding having to do it in every test.

**Review by petebacondarwin — APPROVED** [review]

File: packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts (hunk 1 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/helpers/msw/handlers/deployments.ts:52]: 👍🏻

File: packages/wrangler/src/__tests__/helpers/msw/handlers/script.ts (hunk 2 above)
Comment (JacobMGEvans) [reply packages/wrangler/src/__tests__/helpers/msw/handlers/script.ts:33]: Done.

**Pull request merged**